src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -592,8 +592,12 @@ protected override void ProcessRecord()
                                 OutFile = null;
                             }
 
+                            // Skip insecure redirection detection if the URIs are relative 
+                            bool originIsHttps = response.RequestMessage.RequestUri.IsAbsoluteUri && response.RequestMessage.RequestUri.Scheme == "https";
+                            bool destinationIsHttp = response.Headers.Location is not null && response.Headers.Location.IsAbsoluteUri && response.Headers.Location.Scheme == "http";
+
                             // Detect insecure redirection
-                            if (!AllowInsecureRedirect && response.RequestMessage.RequestUri.Scheme == "https" && response.Headers.Location?.Scheme == "http")
+                            if (!AllowInsecureRedirect && originIsHttps && destinationIsHttp)
```

Comment:
Perhaps makes sense to put "if (!AllowInsecureRedirect" on first place and rest to nested block.
```c#
if (!AllowInsecureRedirect)
{
    bool originIsHttps = response.RequestMessage.RequestUri.IsAbsoluteUri && response.RequestMessage.RequestUri.Scheme == "https";
    bool destinationIsHttp = response.Headers.Location is not null && response.Headers.Location.IsAbsoluteUri && response.Headers.Location.Scheme == "http";

    ...
}

Reply:
done